risc0/zkvm/src/prove/tests.rs

```diff
@@ -169,13 +169,13 @@ fn memory_io() {
     run_memio(&[(POS, 1)]).unwrap();
 
     // Unaligned write is bad
-    run_memio(&[(POS + 1001, 1)]).unwrap_err();
+    run_memio(&[(POS + 1001, 1)]).unwrap();
```

Comment:
So this seems like we need a new way to test this. I feel like we should check that this indeed was a fault.